src/lib/middleware/featureFlagMiddleware.ts

```diff
@@ -0,0 +1,47 @@
+import { NextFunction } from "express"
+import { ArtsyRequest, ArtsyResponse } from "./artsyExpress"
+import { createFlagProvider } from "lib/featureFlags/featureFlags"
+import {
+  FeatureFlagConfig,
+  FeatureFlagProvider,
+} from "lib/featureFlags/featureFlagProvider"
+
+export function featureFlagMiddleware<T extends FeatureFlagConfig>(
+  providerType: symbol,
+  config?: T
+) {
+  let provider
+  return (req: ArtsyRequest, res: ArtsyResponse, next: NextFunction) => {
+    new Promise<FeatureFlagProvider>(async (resolve, reject) => {
+      if (provider) {
+        resolve(provider)
+      }
+      try {
+        provider = await createFlagProvider(providerType, config)
+        resolve(provider)
+      } catch {
+        reject("An unknown error occurred while creating the flag provider.")
+      }
+    }).then(provider => {
+      // Create feature flag context per request
+      const featureFlagContext = {
+        userId: res.locals.user ? res.locals.user.id : null,
+        sessionsId: res.locals.sd.SESSION_ID,
```

Comment:
0. Typo: ~`sessionsId`~ `sessionId`

1. My understanding is that the _client-side_ page will always have an "anonymous id" generated by Segment. That isn't available here. I _think_ Force always tracks a session ID so this might be a reasonable replacement for it (but someone else should confirm).

2. These basic attributes should support the built-in "stickiness" behaviors necessary for per-user toggles or gradual roll-outs or split tests. Additional attributes are possible, but I believe they must be [supported by clients](https://docs.getunleash.io/advanced/custom_activation_strategy) before they can be used, so no rush there.